CMakeLists.txt

```diff
@@ -218,6 +218,24 @@ if(OCOS_BUILD_ANDROID)
   endif()
 
   set(OCOS_BUILD_JAVA ON CACHE INTERNAL "")
+
+  # Add 16KB page size compatibility for Android 15+ devices
+  # See: https://developer.android.com/guide/practices/page-sizes
+  # This flag ensures binaries work on both 4KB and 16KB page size systems
+  option(ANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES "Enable 16KB page size support for Android" ON)
+
+  if(ANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES)
+    # Android NDK uses Clang + lld which supports this flag
+    if(CMAKE_C_COMPILER_ID MATCHES "Clang|GNU")
+      set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-z,max-page-size=16384")
```

Comment:
is it necessary to specify the link options in this variable and also with `target_link_options()`?